lib/Cake/Test/Case/Controller/Component/CookieComponentTest.php

```diff
@@ -790,9 +752,99 @@ public function testDeleteChildrenNotExist() {
 	}
 
 /**
+ * Test that deleting a child entry of the cookie array works
+ *
+ * @return void
+ */
+	public function testDeleteChildEntry() {
+		$response = $this->Controller->response;
+
+		$this->Cookie->write('User.name', 'user', false);
+		$this->Cookie->write('User.email', 'user@example.com', false);
```

Comment:
> Also, from the viewpoint of security, it would be better not to expose any keys of the cookie if it is encrypted.

Agreed. I think that is another drawback to these changes.

> As another approach, we can store expiration information to the cookie.

This could work well. It could get complicated with deeply nested values. Would it be an option to raise an error when a nested value has an expiration set?